application.py

```diff
@@ -94,12 +97,32 @@ def auth():
     else:
         return redirect(url_for('register',args=4))
 
-# @app.route("/search")
-# def search():
-#     if session["Username"]!=None:
-#         return "Maintained successfully"
-#     else: 
-#         return redirect(url_for("logout"))
+@app.route("/search", methods=['POST','GET'])
+def search():
+    if request.method=='POST':
+        field=((request.form['Choose']))
+        key=request.form.get('Search')
+        #print(key)
+        search="%{}%".format(key)
+        if field=="isbn":
+            list=db.query(Book).filter(Book.isbn.like(search))
+            for x in list:
+                return render_template('search.html',list=list)
+        elif field=="title":
+            list=db.query(Book).filter(Book.title.like(search))
+            for x in list:
+                return render_template('search.html',list=list)
+        elif field=="author":
+            list=db.query(Book).filter(Book.author.like(search))
+            for x in list:
+                return render_template('search.html',list=list)
+        else:
+            list=db.query(Book).filter(Book.year.like(search))
+            for x in list:
+                return render_template('search.html',list=list)
```

Comment:
Impliment this in a function and pass the type of search and searched query. Based on the results from the function ,make the routing as per your implimentation. But the function should be written in a separate file and imported into application.py (similar to the LRU and LRU_test)